packages/breadcrumb/src/types.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { HTMLProps, ReactNode } from 'react';
+
+export interface IUseBreadcrumbReturnValue {
+  getContainerProps: <T extends Element>(
+    props: Omit<HTMLProps<T>, 'aria-label' | 'role'> & {
+      'aria-label': NonNullable<HTMLProps<T>['aria-label']>;
+      role?: 'navigation' | null;
+    }
+  ) => HTMLProps<T>;
+  getCurrentPageProps: <T extends Element>(props?: HTMLProps<T>) => HTMLProps<T>;
+}
+
+export interface IBreadcrumbContainerProps {
+  /**
+   * Provides breadcrumb render prop functions
+   *
+   * @param {function} [options.getContainerProps] Container props getter
+   * @param {function} [options.getCurrentPageProps] Current page props getter
+   */
+  render?: (options: {
+    getContainerProps: IUseBreadcrumbReturnValue['getContainerProps'];
+    getCurrentPageProps: IUseBreadcrumbReturnValue['getCurrentPageProps'];
+  }) => ReactNode;
```

Comment:
This pattern was adopted in favor of Storybook readability. Storybook doesn't seem able to navigate the general redirect to produce helpful documentation. Therefore, the code is oriented around informing the consumer rather than writing the most optimal lines-of-code. I wish this could be different, but having tried _many_ different approaches, I think this works in the best interest of the end user.